crates/notary/client/src/client.rs

```diff
@@ -485,6 +474,26 @@ fn default_root_store() -> RootCertStore {
     root_store
 }
 
+// Checks whether the error is potentially related to a mismatch in TLS
+// configuration between the client and the server.
+fn is_tls_mismatch_error(err: &std::io::Error) -> bool {
+    if let Some(rustls::Error::InvalidMessage(rustls::InvalidMessage::InvalidContentType)) = err
+        .get_ref()
+        .and_then(|inner| inner.downcast_ref::<rustls::Error>())
+    {
+        return true;
+    } else if let Some(err) = err
+        .get_ref()
+        .and_then(|inner| inner.downcast_ref::<hyper::Error>())
+    {
+        // Check if the hyper error is `Kind::Parse(Parse::Version)`.
+        if err.to_string() == "invalid HTTP version parsed" {
```

Comment:
Sorry I just realised that `Kind::Parse(Parse::Version)` is not triggered at [`request_notarization()`](https://github.com/tlsnotary/tlsn/blob/de7a47de5b4f6e4de0ec2ff3d1c42a0d2bf7a6a6/crates/notary/client/src/client.rs#L187); rather at [`send_request()`](https://github.com/tlsnotary/tlsn/blob/de7a47de5b4f6e4de0ec2ff3d1c42a0d2bf7a6a6/crates/notary/client/src/client.rs#L298)

Reply:
Anyway that doesn't matter, since these error messages shouldn't be relied upon according to [hyper's doc](https://github.com/hyperium/hyper/blob/0369f5e86406abbd94091ee5120181632c5d98d2/src/error.rs#L20-L22). The only other way I know is
```rust
if format!("{:?}", err).contains("Parse(Version)")
```
which also is kinda brittle.

Given that, I am now leaning towards not catching this error in the client side anymore, since it will be caught in notary server. Let me know what you think.